### PR TITLE
Fix buffer leak with go awaway packets

### PR DESCRIPTION
**Pull request by bondj**
When processing Http2GoAwayEvents Netty reports buffer leaks

## Description
The buffer does not appear to be released, and I think it's acceptable to convert to a string without bumping the refCount

## Motivation and Context
Potential memory leak

## Testing
Currently performing long run test. The leak warnings usually show up in a couple of days, so will confirm a good fix then.
UnitTests updated

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [X ] I have read the **CONTRIBUTING** document
- [ X] Local run of `mvn install` succeeds
- [ X] My code follows the code style of this project
- [ X] My change requires a change to the Javadoc documentation
- [ X] I have updated the Javadoc documentation accordingly
- [ X] I have read the **README** document
- [ X] I have added tests to cover my changes
- [ X] All new and existing tests passed
- [ X] A short description of the change has been added to the **CHANGELOG**
- [ X] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ X] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java [hunk 1]
```diff
@@ -41,7 +42,7 @@ public Http2GoAwayEventListener(Channel parentChannel) {
     @Override
     public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
         Http2MultiplexedChannelPool channelPool = parentChannel.attr(ChannelAttributeKey.HTTP2_MULTIPLEXED_CHANNEL_POOL).get();
-        GoAwayException exception = new GoAwayException(errorCode, debugData.retain());
+        GoAwayException exception = new GoAwayException(errorCode, debugData.toString(StandardCharsets.UTF_8));
```
Comment (zoewangg) [inline http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java:45]: Should we release the ByteBuf after this? Preferably in a finally block?

**Review by zoewangg — COMMENTED** [review]
Thanks for the PR! Could you add a changelog entry? You can generate one by running the `scripts/new-change` script and following the instructions.

File: http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java (hunk 1 above)
Comment (bondj) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java:45]: I do not believe so... The original bug was that by retaining the debugData, the refCount was increased, but never released.  I restructured the code, to convert to a String and pass into the GoAwayException, so we always know the lifecycle of that bytebuf, and it seemed tidier than try/final the modifying of the refCount for processing that happens all within this method..

The Javadoc for https://netty.io/4.1/api/io/netty/handler/codec/http2/DefaultHttp2Connection.html goAwayReceived says 
> debugData - The Additional Debug Data in the GOAWAY frame. Note that reference count ownership belongs to the caller (ownership is not transferred to this method).
So I think it's safe to just use the debugData content within the scope of the method, and the code shouldn't release it.

I will do a quick check, to see if this bug appears elsewhere.

Comment (bondj) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java:45]: I will update the changelog once we've run the fix for a week, and I'm confident that it fixes our problem :)

Comment (bondj) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java:45]: Looks like the fix, fixes the problem

Comment (zoewangg) [reply http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2GoAwayEventListener.java:45]: Nice, thanks for verifying.

**Review by zoewangg — APPROVED** [review]

**Comment by zoewangg** [comment]
Thank you again for your contribution!

@all-contributors please add @bondj for code

**Pull request merged**